model/spectrum.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * spectrum.h
+ */
+#ifndef DXTBX_MODEL_SPECTRUM_H
+#define DXTBX_MODEL_SPECTRUM_H
+
+#include <iostream>
+#include <cmath>
+#include <scitbx/vec3.h>
+#include <scitbx/array_family/shared.h>
+#include <scitbx/array_family/simple_io.h>
+#include <scitbx/array_family/simple_tiny_io.h>
+#include <dxtbx/error.h>
+#include "model_helpers.h"
+
+namespace dxtbx { namespace model {
+  typedef scitbx::af::shared<double> vecd;
+
+  /** A class to represent a 2D spectrum. */
+  class Spectrum {
+  public:
+    /** Default constructor: initialise all to zero */
+    Spectrum() {}
+
+    /**
+     * Initialise all the spectrum parameters.
+     * @param energies The spectrum energies (eV)
+     * @param energies The spectrum weights (unitless)
+     */
+    Spectrum(vecd energies, vecd weights)
+        : energies_(energies),
+          weights_(weights) {}
+
+    virtual ~Spectrum() {}
+
+    /* Get the spectrum energies (eV) */
+    vecd get_energies() const {
+      return energies_;
+    }
+
+    /* Get the spectrum weights (unitless) */
+    vecd get_weights() const {
+      return weights_;
+    }
+
+    double get_weighted_wavelength() const {
```

Comment:
If we also had `get_weighted_wavelength_variance` then one could _almost_ model a Gaussian beam with this 🤔

Reply:
Also please can has `get_weighted_energy_eV` and it's `variance` equivalent as part of the API - we may not use this but we should probably have it. 

This could be a simple refactor as keeping most of the code in this current method as `get_weighted_energy_eV()` and then converting the result to a wavelength before returning, for consistency.